core/common/src/main/kotlin/app/k9mail/core/common/mail/EmailAddressParserConfig.kt

```diff
@@ -3,39 +3,63 @@ package app.k9mail.core.common.mail
- * instead of retaining the original user input.
+ * @param isLocalPartRequiringQuotedStringAllowed Email addresses whose local part requires the use of a quoted string
+ * are only allowed when this is `true`. This is separate from [isQuotedLocalPartAllowed] because one might want to
+ * allow email addresses that unnecessarily use a quoted string, e.g. `"test"@domain.example`
+ * ([isQuotedLocalPartAllowed] = `true`, [isLocalPartRequiringQuotedStringAllowed] = `false`; [EmailAddress] will not
+ * retain the original form and treat this address exactly like `test@domain.example`). When allowing this, remember to
+ * use the value of [EmailAddress.address] instead of retaining the original user input.
  *
- * The value of this property is ignored if [allowQuotedLocalPart] is `false`.
+ * The value of this property is ignored if [isQuotedLocalPartAllowed] is `false`.
  *
- * @param allowEmptyLocalPart Email addresses with an empty local part (e.g. `""@domain.example`) are only allowed if
- * this value is `true`.
+ * @param isEmptyLocalPartAllowed Email addresses with an empty local part (e.g. `""@domain.example`) are only allowed
+ * if this value is `true`.
  *
- * The value of this property is ignored if at least one of [allowQuotedLocalPart] and
- * [allowLocalPartRequiringQuotedString] is `false`.
+ * The value of this property is ignored if at least one of [isQuotedLocalPartAllowed] and
+ * [isLocalPartRequiringQuotedStringAllowed] is `false`.
  */
 data class EmailAddressParserConfig(
-    val allowQuotedLocalPart: Boolean,
-    val allowLocalPartRequiringQuotedString: Boolean,
-    val allowEmptyLocalPart: Boolean = false,
+    val isLocalPartLengthCheckEnabled: Boolean,
+    val isEmailAddressLengthCheckEnabled: Boolean,
+    val isQuotedLocalPartAllowed: Boolean,
+    val isLocalPartRequiringQuotedStringAllowed: Boolean,
+    val isEmptyLocalPartAllowed: Boolean = false,
 ) {
     companion object {
         /**
-         * This configuration should match what `EmailAddressValidator` currently allows.
+         * This allows local parts requiring quoted strings and disables length checks for the local part and the
+         * whole email address.
+         */
+        val RELAXED = EmailAddressParserConfig(
+            isLocalPartLengthCheckEnabled = false,
+            isEmailAddressLengthCheckEnabled = false,
+            isQuotedLocalPartAllowed = true,
+            isLocalPartRequiringQuotedStringAllowed = true,
+            isEmptyLocalPartAllowed = false,
+        )
+
+        /**
+         * This only allows a subset of valid email addresses. Use this when validating the email address a user wants
+         * to add to an account/identity.
          */
-        val DEFAULT = EmailAddressParserConfig(
-            allowQuotedLocalPart = true,
-            allowLocalPartRequiringQuotedString = true,
-            allowEmptyLocalPart = false,
+        val LIMITED = EmailAddressParserConfig(
```

Comment:
I think this could be named `STRICT`, limited sounds like reduced funtionality

Reply:
It kind of is, because we don't allow email addresses with a quoted local part, e.g. `"foo bar"@domain.example`.